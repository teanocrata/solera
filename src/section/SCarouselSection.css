.carousel-inner div {
  padding-top: 30px;
  display: flex;
  flex-direction: column;
  align-items: center;
}
.carousel-inner img {
  max-width: 250px;
  max-height: 250px;
}

.carousel-control.right, .carousel-control.left {
    color: forestgreen;
}

.carousel-indicators li.active{
  background-color: forestgreen;
}
