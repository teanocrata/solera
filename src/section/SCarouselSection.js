import React, { Component } from "react";
import { Carousel, Image } from "react-bootstrap";
import campera from "./campera_square.jpg";
import pixie from "./pixie_square.jpg";
import cheyenne from "./cheyenne_square.jpg";

import "./SCarouselSection.css";

class SCarouselSection extends Component {
  render() {
    const style = {
      image: {
        boxShadow: "0 8px 16px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19)"
      }
    };
    return (
      <Carousel>
        <Carousel.Item>
          <div
            style={{
              height: 500,
              width: "100%",
              backgroundColor: "transparent"
            }}
          >
            <Image style={style.image} src={cheyenne} circle />
          </div>
          <Carousel.Caption>
            <h3>Cheyenne</h3>
            <p>
              Uno de nuestros caballos estrella, muy flexible, con doma western
              y clásica. Un excelente saltador aunque cuidado, ¡a veces tiene un
              carácter...!
            </p>
          </Carousel.Caption>
        </Carousel.Item>
        <Carousel.Item>
          <div
            style={{
              height: 500,
              width: "100%",
              backgroundColor: "transparent"
            }}
          >
            <Image style={style.image} src={pixie} circle />
          </div>
          <Carousel.Caption>
            <h3>Pixie</h3>
            <p>
              Un caballo pequeñito, casi un pony, super noble, tranquilo y
              adaptándose a todas las situaciones, tanto en pista como en el
              campo. Muy divertido y juguetón.
            </p>
          </Carousel.Caption>
        </Carousel.Item>
        <Carousel.Item>
          <div
            style={{
              height: 500,
              width: "100%",
              backgroundColor: "transparent"
            }}
          >
            <Image style={style.image} src={campera} circle />
          </div>
          <Carousel.Caption>
            <h3>Campera</h3>
            <p>
              Especialista en rutas por el campo, ¡es enorme!, al principio
              impone un poco pero es un pedazo de pan. En la pista es
              espectacular, eso si, tienes que estar bien en forma ;)
            </p>
          </Carousel.Caption>
        </Carousel.Item>
      </Carousel>
    );
  }
}

export default SCarouselSection;
