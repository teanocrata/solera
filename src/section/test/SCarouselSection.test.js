import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import SCarouselSection from "../SCarouselSection";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SCarouselSection />, div);
});

test("SCarouselSection renders correctly", () => {
  const section = renderer.create(<SCarouselSection />).toJSON();
  expect(section).toMatchSnapshot();
});

test("SCarouselSection renders correctly with bodies", () => {
  const bodies = [
    {
      title: "Cheyenne",
      image: {
        src: "https://drive.google.com/uc?id=0BwBkl30bqN8KQUZselFVQmh3cHM"
      },
      text:
        "  Uno de nuestros caballos estrella, muy flexible, con doma western y clásica. Un excelente saltador aunque cuidado, ¡a veces tiene un carácter...!"
    }
  ];
  const section = renderer
    .create(<SCarouselSection bodies={bodies} />)
    .toJSON();
  expect(section).toMatchSnapshot();
});
