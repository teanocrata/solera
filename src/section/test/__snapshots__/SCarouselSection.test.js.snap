// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCarouselSection renders correctly 1`] = `
<div
  className="carousel slide"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <ol
    className="carousel-indicators"
  >
    <li
      className="active"
      onClick={[Function]}
    />
     
    <li
      className={null}
      onClick={[Function]}
    />
     
    <li
      className={null}
      onClick={[Function]}
    />
     
  </ol>
  <div
    className="carousel-inner"
  >
    <div
      className="item active"
    >
      <div
        style={
          Object {
            "backgroundColor": "transparent",
            "height": 500,
            "width": "100%",
          }
        }
      >
        <img
          className="img-circle"
          src="cheyenne_square.jpg"
          style={
            Object {
              "boxShadow": "0 8px 16px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19)",
            }
          }
        />
      </div>
      <div
        className="carousel-caption"
      >
        <h3>
          Cheyenne
        </h3>
        <p>
          Uno de nuestros caballos estrella, muy flexible, con doma western y clásica. Un excelente saltador aunque cuidado, ¡a veces tiene un carácter...!
        </p>
      </div>
    </div>
    <div
      className="item"
    >
      <div
        style={
          Object {
            "backgroundColor": "transparent",
            "height": 500,
            "width": "100%",
          }
        }
      >
        <img
          className="img-circle"
          src="pixie_square.jpg"
          style={
            Object {
              "boxShadow": "0 8px 16px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19)",
            }
          }
        />
      </div>
      <div
        className="carousel-caption"
      >
        <h3>
          Pixie
        </h3>
        <p>
          Un caballo pequeñito, casi un pony, super noble, tranquilo y adaptándose a todas las situaciones, tanto en pista como en el campo. Muy divertido y juguetón.
        </p>
      </div>
    </div>
    <div
      className="item"
    >
      <div
        style={
          Object {
            "backgroundColor": "transparent",
            "height": 500,
            "width": "100%",
          }
        }
      >
        <img
          className="img-circle"
          src="campera_square.jpg"
          style={
            Object {
              "boxShadow": "0 8px 16px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19)",
            }
          }
        />
      </div>
      <div
        className="carousel-caption"
      >
        <h3>
          Campera
        </h3>
        <p>
          Especialista en rutas por el campo, ¡es enorme!, al principio impone un poco pero es un pedazo de pan. En la pista es espectacular, eso si, tienes que estar bien en forma ;)
        </p>
      </div>
    </div>
  </div>
  <a
    className="carousel-control left"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <span
      className="glyphicon glyphicon-chevron-left"
    />
    <span
      className="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    className="carousel-control right"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <span
      className="glyphicon glyphicon-chevron-right"
    />
    <span
      className="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
